motr/io_req_fop.c

```diff
@@ -505,8 +505,6 @@ static void ioreq_cc_bottom_half(struct m0_sm_group *grp,
 		ti->ti_rc = rc;
 	if (xfer->nxr_rc == 0 && rc != 0)
 		xfer->nxr_rc = rc;
-	if (ioo->ioo_rc == 0 && rc != 0)
-		ioo->ioo_rc = rc;
```

Comment:
I did some tests for more than k disk failures: for a cluster configuration of 5 ioses, 10 disks (2 per ios), 6+2+0:

(1) if one ioservice is killed, the degraded works, if 2 ioservices are killed (1 service tolerance), degraded write fails. These are expected behaviours, right?

(2)  If I setup bad disks (4 disks) with dmsetup and the bad disks start returning errors after writing objects, motr degraded write fails as there are  more than 2 disk fails when writing a parity group.
 
(3) If I setup bad disks (3 disks), looks like motr manages to choose unit's target disks so there are at most 2 failed disks in a parity group in my tests, because of it the degrade write goes ahead, put-obj successes. I tested multiple 200MB and 1GB objects, all successes ( note: from the m0trace, motr didn't switch to alternative pool version). Will looks further to understand why it works so.

@andriytk any other cases you want me to test?

Reply:
Just to confirm: if it fails to create more than K cobs - it fails (as it should be), right?

Reply:
@andriytk I emulated errors for cob creation, and can confirm that write fails if more than K cob creations fail.

Reply:
Thank you.